compliance_checker/cf/cf.py

```diff
@@ -3698,6 +4035,286 @@ def check_cell_measures(self, ds):
+        :return two-tuple of (bool, str)
+        """
+
+        return (
+            val in horizontal_datum_names17,
+            ('{} must be a valid Horizontal Datum Name; '
+
+             'see https://github.com/cf-convention/cf-conventions/wiki/Mapping-from-CF-Grid-Mapping-Attributes-to-CRS-WKT-Elements.')
+        )
+
+    def _evaluate_prime_meridian_name(self, val):
+        """
+        Evaluate the condition for the prime_meridian_name.
+
+        :param val: value to be tested
+        :rtype tuple
+        :return two-tuple of (bool, str)
+        """
+
+        return (
+            val in prime_meridian_names17,
+            ('{} must be a valid Prime Meridian name; '
+             'see https://github.com/cf-convention/cf-conventions/wiki/csv/prime_meridian.csv.')
+        )
+
+    def _evaluate_projected_crs_name(self, val):
+        """
+        Evaluate the condition for the projected_crs attribute.
+
+        :param val: value to be tested
+        :rtype tuple
+        :return two-tuple of (bool, str)
+        """
+
+        conn = self._get_projdb_conn()
+        query_str = ('SELECT 1 FROM projected_crs WHERE name = ? ' 
+                     'UNION ALL '
+                     'SELECT 1 FROM alias_name WHERE alt_name = ? '
+                     'AND table_name = \'projected_crs\' LIMIT 1')
+
+        # try to find the value in the database
+        res_set = conn.execute(query_str, (val, val))
+
+        # does it exist? if so, amt returned  be > 1
+        res = len(res_set.fetchall()) > 0
+
+        return (
+            res,
+            'projected_crs_name must correspond to a valid OGC WKT PROJCS name'
+        )
+
+    def _evaluate_reference_ellipsoid_name(self, val):
+        """
+        Evaluate the condition for the reference_ellipsoid_name attribute.
+
+        :param val: value to be tested
+        :rtype tuple
+        :return two-tuple of (bool, str)
+        """
+
+        return (
+            val in ellipsoid_names17,
+            ('{} must be a valid Ellipsoid Name; '
+             'see https://github.com/cf-convention/cf-conventions/wiki/csv/ellipsoid.csv.')
+        )
+
+    def _evaluate_towgs84(self, val):
+        """
+        Evaluate the condition for the towgs84 attribute.
+
+        :param val: value to be tested
+        :rtype tuple
+        :return two-tuple of (bool, str)
+        """
+
+        msg = ('towgs84 must be an array of length 3, 6, or 7 of double-precision'
+               ' and correspond to anm OGC WKT TOWGS84 node')
+
+        # if not numpy type, return false
+        if not getattr(val, 'dtype', None):
+            return (False, msg)
+
+        # must be double-precision array
+        elif val.dtype != np.float64:
+            return (False, msg)
+
+        # must be of length 3, 6, or 7
+        elif not val.shape: # single value
+            return (False, msg)
+
+        elif not any([(val.shape[0] == 3) or (val.shape[0] == 6) or (val.shape[0] == 7)]):
```

Comment:
Using `val.size` may be somewhat easier here -- array should probably only have one dimension/be a vector.   Also use something like `val.size in (3, 6, 7)` to abridge the chained or statements.